pkg/pipeline/stream/stream_test.go

```diff
@@ -301,15 +321,31 @@ func generatorSource(ctrl *gomock.Controller, logger log.CtxLogger, nodeID strin
 }
 
 func printerDestination(ctrl *gomock.Controller, logger log.CtxLogger, nodeID string) connector.Destination {
+	rchan := make(chan record.Record)
 	destination := connmock.NewDestination(ctrl)
 	destination.EXPECT().Open(gomock.Any()).Return(nil).Times(1)
-	destination.EXPECT().Teardown(gomock.Any()).Return(nil).Times(1)
 	destination.EXPECT().Write(gomock.Any(), gomock.Any()).DoAndReturn(func(ctx context.Context, r record.Record) error {
 		logger.Debug(ctx).
 			Str("node_id", nodeID).
 			Msg("got record")
+		rchan <- r
 		return nil
 	}).AnyTimes()
+	destination.EXPECT().Ack(gomock.Any()).DoAndReturn(func(ctx context.Context) (record.Position, error) {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		case r, ok := <-rchan:
+			if !ok {
+				return nil, nil
+			}
+			return r.Position, nil
+		}
+	}).AnyTimes()
+	destination.EXPECT().Teardown(gomock.Any()).DoAndReturn(func(ctx context.Context) error {
+		close(rchan)
+		return nil
+	}).Times(1)
```

Comment:
Now we actually need to make sure our mock is returning acks for records correctly so that they are received by `AckerNode` (same logic had to be added to mocks in `lifecycle_test.go`).